cache/cache.go

```diff
@@ -27,34 +27,61 @@ const (
 	LevelAll  = Level("all")
 )
 
-// Collect walks the directory tree underneath projectRoot and registers matching
-// paths for caching based on the value of cacheLevel. Returns an error if there
-// was an underlying error that would lead to a corrupted cache file, otherwise
-// the given path is skipped.
-func Collect(projectRoot string, cacheLevel Level) error {
+// ItemCollector ...
+type ItemCollector interface {
```

Comment:
NOTE: Long term it would be better to have a generic cache interface in go-steputils.

Reply:
Moved CacheLevel and ItemCollecto to go-steputils/cache: https://github.com/bitrise-io/go-steputils/pull/31